internal/assertions/apiversions_response_assertion.go

```diff
@@ -23,96 +23,75 @@ var errorCodes = map[int]string{
 type ApiVersionsResponseAssertion struct {
 	ActualValue   kafkaapi.ApiVersionsResponse
 	ExpectedValue kafkaapi.ApiVersionsResponse
-	logger        *logger.Logger
-	err           error
```

Comment:
We don't need `err` as state either, it's just returned from `Run()` and bubbled up from the assert calls